listenbrainz/webserver/views/api_tools.py

```diff
@@ -49,6 +52,8 @@ def insert_payload(payload, user, listen_type=LISTEN_TYPE_IMPORT):
         _send_listens_to_queue(listen_type, augmented_listens)
     except (APIInternalServerError, APIServiceUnavailable) as e:
         raise
+    except DataError:
+        raise APIBadRequest("Listen submission contains invalid characters.")
```

Comment:
sorry to be picky here - are we sure that DataError is raised only if data is invalid? what does "invalid" mean? 
I see that we made a similar wishy-washy statement in AB, so maybe it's not a huge issue.

We should explicitly say in the documentation that null characters are not allowed. An alternate way to do this check would be to additionally have a check the moment that we load the POST data, and look for null characters, and if so return APIBadRequest immediately with an error message that explicitly mentions these bad characters

Reply:
I would prefer to handle this after the insert fails. Sentry had ~50k of these messages even if we assume those were all unique (no retries) its hardly 0.1% of the submissions. Performing a check on all data to catch less than 0.1% sounds a bit overkill to me especially since the bad data is harmless.

I do agree with the DataError and message thing. I'll look further into it and see if we can improve it further.